task2/rest-api-basics/web/src/main/java/com/epam/esm/web/config/WebAppConfig.java

```diff
@@ -0,0 +1,73 @@
+package com.epam.esm.web.config;
+
+import com.epam.esm.converter.impl.GiftCertificateConverter;
+import com.epam.esm.converter.impl.TagConverter;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.*;
+import org.springframework.core.env.Environment;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.jdbc.support.JdbcTransactionManager;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.sql.DataSource;
+
+@Configuration
+@EnableTransactionManagement
```

Comment:
Each of your modules(except domain) should include its own **Configuration** class. Each of these configurations has to make component scan only for this module and customize appropriate beans. In order to include configuration from one module to another one, use annotation @Import.

Reply:
thanks